### ISSUE TITLE
Decide what an MOC infra cluster would look like (software, hardware)

### ISSUE THREAD

**Issue by hakasapl** [CCI-MOC/ops-issues#1150]

**Comment by naved001** [comment]
I was thinking about the R640s for this and not R440.

**Comment by hakasapl** [comment]
Right, I forgot about those. So we will move Netbox from ocp-prod to ovirt temporarily, then we will down ocp-prod and use those 3 controllers as hybrid controller/workers.

In addition, I will add SSDs to these nodes

**Comment by larsks** [comment]
There are a few things we should consider here:

## Virtualization and cluster size

> Because we need to run both containers and VMs...

Have we identified specific virtualized workloads that we want to run on this cluster? Because virtualization tends to consume more memory than containerized services, if we expect to be using a lot of virtualization we may want to start with something other than a minimal cluster.

## Storage

What are we thinking in terms of storage? I would argue that we don't want to depend on an external storage cluster, given the nature of the workloads we'll be running here. At the same time, we will want the ability to create storage volumes that aren't node-locked.

- Do we run an internal ODF instance? In that case we'll need a chunk of worker nodes just to support the storage cluster.

- What about something like [Longhorn](https://longhorn.io/)? If the nodes have a chunk of internal storage that may be a viable -- although unsupported -- option.

- Or maybe a small external ceph cluster that is dedicated to this infra cluster?

**Comment by hakasapl** [comment]
> What about something like Longhorn? If the nodes have a chunk of internal storage that may be a viable -- although unsupported -- option.

Unsupported by who? Red hat/openshift?

How would we go about block storage with Longhorn? If we still need an additional block storage source I'm tempted to either go the ODF or ceph route since we should be able to provide both with those.

@larsks

**Comment by larsks** [comment]
> Unsupported by who? Red hat/openshift?

By anyone, as far as I can tell, unless Rancher offers support contracts.

> How would we go about block storage with Longhorn?

Longhorn exposes block storage that is directly attached to nodes. The idea here was that we could have a cluster with minimal external dependencies. By fully populating drive bays in our servers we should be able to get multiple TB of storage which ought to be more than sufficient.

If folks would prefer to use ODF with an external ceph cluster, I'm won't complain too much :).

**Comment by hakasapl** [comment]
Longhorn sounds good - I agree that the less dependencies the better for this cluster. @naved001 any thoughts?

**Comment by naved001** [comment]
Let's go with Longhorn. The fewer the dependencies, the better.

**Comment by joachimweyl** [comment]
@hakasapl can you update the description with the changes and decisions made in the comments so we have a comprehensive understanding of what we are going to do at the top? Do we have any more decisions to make or is it time to create issues and maybe an epic around the creation of this cluster?

**Comment by hakasapl** [comment]
Done

**Comment by joachimweyl** [comment]
@hakasapl is the plan to create an epic or issue to track what needs to happen next for the MOC Infra Cluster?

**Comment by hakasapl** [comment]
I've made an epic with this info: https://app.zenhub.com/workspaces/moc-a--rh-sprint-workspace-62a210f69d42f600151deae0/epics/Z2lkOi8vcmFwdG9yL1plbmh1YkVwaWMvMTY4NDY2?state=OPEN&state=TODO&state=IN_PROGRESS
